app/services/core/service.ts

```diff
@@ -90,7 +91,14 @@ export abstract class Service {
 
   static get instance() {
     const instance = !this.hasInstance ? Service.createInstance(this) : this[singleton];
-    return this.proxyFn ? this.proxyFn(instance) : instance;
+
+    if (this.proxyFn) {
+      if (!proxies[this.name]) proxies[this.name] = this.proxyFn(instance);
```

Comment:
i just noticed that `proxies` is an array but here we key into it like an object, i presume that's not intentional?